### PR TITLE
add branch-validation role; modify branch-validation.yml playbook

### PR DESCRIPTION
**Pull request by angstwad**
Moves branch-validation playbook tasks into a role that has a dependency on the
endpoints role.

### CODE REVIEW

**Comment by j2sol** [comment]
retest

**Comment by angstwad** [comment]
retest

File: roles/branch-validation/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      project_name: admin
+      password: "{{ secrets.admin_password }}"
+      auth_url: "http://127.0.0.1:{{ endpoints.keystone_admin.port.backend_api }}/{{ endpoints.keystone.version }}"
+    name: validate
+    state: present
+  register: keypair
+
+- name: persist private key
+  sudo: no
+  local_action:
```
Comment (j2sol) [inline roles/branch-validation/tasks/main.yml]: As a style, we like `delegate_to: localhost`

Comment (j2sol) [inline roles/branch-validation/tasks/main.yml]: Also, as discussed, we do want to do this from the controller, it's not always the case that the API endpoints will be available to the bastion.

File: roles/branch-validation/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      auth_url: "http://127.0.0.1:{{ endpoints.keystone_admin.port.backend_api }}/{{ endpoints.keystone.version }}"
+    network: external
+    server: "validate-{{ groups[instance_group][0] }}"
+    reuse: true
+  register: fip
+  ignore_errors: yes
+
+- name: wait for instances to be reachable
+  sudo: no
+  local_action:
```
Comment (j2sol) [inline roles/branch-validation/tasks/main.yml]: `delegate_to`

File: roles/branch-validation/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    module: wait_for
+    port: 22
+    delay: 5
+    timeout: 300
+    host: "{{ fip.floating_ip.floating_ip_address }}"
+
+- name: add host
+  local_action:
```
Comment (j2sol) [inline roles/branch-validation/tasks/main.yml]: `delegate_to`

**Comment by masteinhauser** [comment]
retest

**Pull request merged**